libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -645,6 +657,34 @@ const AP_Param::GroupInfo Compass::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO("CUS_YAW", 51, Compass, _custom_yaw, 0),
 #endif
+
+    // @Param: AD_MAX
+    // @DisplayName: Compass maximum Angle-difference (AD) in all axis (xyz plane)
+    // @Description: This sets the maximum allowed angle difference in all axis between the two compass' for calibration check and arming consistency checks
+    // @Range: -180 180
+    // @Units: deg
+    // @Increment: 1
+    // @User: Advanced
+    AP_GROUPINFO("AD_MAX", 52, Compass, _ang_diff_xyz_max, AP_COMPASS_ANG_DIFF_XYZ_MAX),
```

Comment:
Instead of predirectives, can the defaults be set directly?

Reply:
I prefer keeping it consistent with the rest of the file, such that we use predirectives rather than constant if they are defined. Case in point:
- Line 489: AP_GROUPINFO("OFFS_MAX", 31, Compass, _offset_max, AP_COMPASS_OFFSETS_MAX_DEFAULT)
-  Line 510: AP_GROUPINFO("FLTR_RNG", 34, Compass, _filter_range, HAL_COMPASS_FILTER_DEFAULT)
etc.
Although I can add a suffix '_DEFAULT' to make it consistent to better readability and maintain homogeneousity throughout.

Reply:
Is it maintaining consistency if there are 55 parameters and 4 have preprocessor directives?
Keeping the default value here prevents the need to scroll up to check the value.

Reply:
By consistency I meant, if there are preprocessor directives, so we should be using them as defaults rather than constants. We can always remove them but seems like the 'Max' value params are set preprocessor directives, so I chose to follow the trend and went with how it was for the 'max param', default values.
I agree that removing the predirectives will make the code smaller and make it more readable, it is simply a choice that was made based on how I saw the code to be, I am more than happy to change it if that is what you think would seem more readable.

Reply:
Since these preprocessor directives are not used elsewhere, most defaults don't have them, and finding the value requires scrolling, I recommend removing them

Reply:
I have made the desired changes. Thanks! for the feedback